### PR TITLE
land_detector move orb subscriptions to uORB::Subscription

### PR DESCRIPTION
**Pull request by dagar**
Requires #12040 before merging.

### CODE REVIEW

**Comment by dagar** [comment]
Ready for review and testing.

File: src/modules/land_detector/LandDetector.cpp [hunk 1]
```diff
@@ -46,50 +46,33 @@
...
-		orb_unadvertise(_landDetectedPub);
-	}
 }
 
 int LandDetector::start()
 {
 	/* schedule a cycle to start things */
-	return work_queue(HPWORK, &_work, (worker_t)&LandDetector::_cycle_trampoline, this, 0);
-}
+	ScheduleOnInterval(LAND_DETECTOR_UPDATE_INTERVAL, 10000);
```
Comment (julianoes) [inline src/modules/land_detector/LandDetector.cpp]: You could remove the comment above.

Comment (dagar) [reply src/modules/land_detector/LandDetector.cpp]: Done.

**Comment by jorge789** [comment]
### Tested on Pixhawk 2 Cube V3
**Flight Card 1**

**Modes Tested:**

Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.
**Procedure:**
Arm in position mode, check take off behavior (Vehicle should stay on the ground as long as the throttle stick is at the middle or below, and start to ascend smoothly as long as throttle stick as above the middle.), the land and disarm.
Same procedure was repeated with Altitude and Stabilized mode.

**Notes:**
No issues were noted, good flight in general.

**log**
https://review.px4.io/plot_app?log=a29f37d6-600f-4473-925e-b6b56aa036d3


### Tested on PixRacer V4

**Modes Tested**

Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.
Mission Plan Mode (Automated): Good.
RTL: Good.

**- Procedure**
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoin activate RTL and see landing behaviour.
**- Notes**
No issues noted, good flight in general.

**Logs**
https://review.px4.io/plot_app?log=ad46cac7-4f6d-48f2-84ab-2c35af0d6374

https://review.px4.io/plot_app?log=69b99981-076c-4533-a80d-c30982dc5eea

**Pull request merged**